packages/pug-code-gen/index.js

```diff
@@ -653,27 +889,68 @@ Compiler.prototype = {
    * @api public
    */
 
-  visitCode: function(code){
+  visitCode: function(code, ctx){
     // Wrap code blocks with {}.
     // we only wrap unbuffered code blocks ATM
     // since they are usually flow control
-
     // Buffer code
+    //
+    var ast = [];
     if (code.buffer) {
       var val = code.val.trim();
       val = 'null == (pug_interp = '+val+') ? "" : pug_interp';
       if (code.mustEscape !== false) val = this.runtime('escape') + '(' + val + ')';
-      this.bufferExpression(val);
+      push.apply(ast, this.bufferExpression(val));
     } else {
-      this.buf.push(code.val);
-    }
 
-    // Block support
-    if (code.block) {
-      if (!code.buffer) this.buf.push('{');
-      this.visit(code.block, code);
-      if (!code.buffer) this.buf.push('}');
+      var val = code.val.trim();
```

Comment:
I need to think some more about how we handle unbuffered code blocks, I suspect this may be a big part of where time is spent parsing (we'd need data to confirm though).

Reply:
it was 1 year ago but I remember that I had difficulties with  an unbuffered code block as a child of an unbuffered code block + and unmatched braces

```pugjs
- if (true) {
  - a = 3
- } else {
- }
```

not implementing this was faster from what I remember but it currently works and is untested in pugjs.